src/context/playlists-context/playlists-context.js

```diff
@@ -0,0 +1,172 @@
+import { useContext, useState, createContext } from "react";
+import { useEffect } from "react";
+import { useToast } from "../toast-context/toast-context";
+import axios from "axios";
+
+const PlaylistsContext = createContext();
+
+const usePlaylists = () => useContext(PlaylistsContext);
+
+function PlaylistsProvider({ children }) {
+  const encodedToken = localStorage.getItem("smashTubeToken");
+  const [playlists, setPlaylists] = useState([]);
+  const [playlist, setPlaylist] = useState([]);
+  const { dispatch } = useToast();
+
+  useEffect(() => {
+    (async function () {
+      try {
+        const playlistsResponse = await axios.get("/api/user/playlists", {
+          headers: {
+            authorization: encodedToken,
+          },
+        });
+
+        if (playlistsResponse.status === 200) {
+          setPlaylists(playlistsResponse.data.playlists);
+        }
+      } catch (error) {
+        console.error(error.response.data.errors);
+      }
+    })();
+  }, [encodedToken]);
+
+  const addToPlaylists = async (playlist) => {
```

Comment:
you can name this function as addPlaylist cause you are repeating same function name on line no 10.6